wpgtk/data/files.py

```diff
@@ -55,8 +63,17 @@ def get_cache_path(wallpaper, backend=None):
         backend = settings.get("backend", "wal")
 
     filepath = join(WALL_DIR, wallpaper)
-    filename = cache_fname(filepath, backend, False, WPG_DIR)
+    # placeholder for variable
+    filename = None
 
+    try:
+        filename = cache_fname(filepath, backend, False, WPG_DIR)
+    except TypeError as error:
+        # in pywal16cols this function have another api
+        if __check_is_pywal16cols():
+            # pywal16cols add boolean after `backend` that means 16cols mode or standart wal
+            filename = cache_fname(filepath, backend, True, False, WPG_DIR) 
+        else: raise error
```

Comment:
why not check `__check_is_pywal16cols()` before calling cache_fname instead of letting it fail?

Reply:
There is another error there, the 16 cols mode is no longer a boolean switch, but a string that may be "darken" or "lighten" as pywal16 can generate the additional shades in one of 2 ways.

Darken: which darkens the colors 1 to 7.
Lighten: which lightens colors 9 to 15 (also saturates them by 50%) 

Tho checking that adjust i think i screwed up some and may need to fix it, as it should default to darken tho if the function is passed a boolean it won't do the default of darken... so i'll have to create a 3.5.4 release...